### PR TITLE
Clarify static name attribute in RouterOutlet API doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
Updates API doc for `RouterOutlet`
https://angular.io/api/router/RouterOutlet#description

- [x] Documentation content changes

## What is the current behavior?
Not clear that `name` is a static attribute that cannot be set or changed dynamically.

Issue Number: https://github.com/angular/angular/issues/24720
https://github.com/angular/angular/issues/12522

## What is the new behavior?
Clarifies status of `RouterOutlet` naming.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/router/src/directives/router_outlet.ts [hunk 1]
```diff
@@ -18,14 +18,17 @@ import {PRIMARY_OUTLET} from '../shared';
...
  * ```
  * <router-outlet></router-outlet>
  * <router-outlet name='left'></router-outlet>
  * <router-outlet name='right'></router-outlet>
  * ```
  *
- * A router outlet will emit an activate event any time a new component is being instantiated,
- * and a deactivate event when it is being destroyed.
+ * A router outlet emits an activate event when a new component is instantiated,
+ * and a deactivate event when it is destroyed.
```
Comment (Splaktar) [inline packages/router/src/directives/router_outlet.ts]: Maybe better as "when a component is destroyed."

**Review by Splaktar — APPROVED** [review]

File: packages/router/src/directives/router_outlet.ts [hunk 2]
```diff
@@ -18,14 +18,17 @@ import {PRIMARY_OUTLET} from '../shared';
  *
  * Acts as a placeholder that Angular dynamically fills based on the current router state.
  *
+ * Each outlet can have a unique name, determined by the optional `name` attribute.
```
Comment (trotyl) [inline packages/router/src/directives/router_outlet.ts]: Maybe the doc should mention that `name` attribute defaults to be `"primary"` if not specified?

Comment (Splaktar) [reply packages/router/src/directives/router_outlet.ts]: Agreed. Confirmation [here](https://github.com/angular/angular/blob/1e9eeafa9e3fbfad5e94d697b8079cf649db626e/packages/router/src/shared.ts#L18)

**Review by gkalpak — APPROVED** [review]

**Comment by jbogarthyde** [comment]
Doc only, should not need G3

**Comment by jbogarthyde** [comment]
Doc only, shouldn't need G3

**Review by atscott — APPROVED** [review]

**Comment by jbogarthyde** [comment]
caretaker: docs only

**Pull request closed without merging**